core/app/models/workarea/checkout.rb

```diff
@@ -161,7 +161,14 @@ def continue_as(user)
     # @return [self]
     #
     def update(params = {})
-      steps.each { |s| s.new(self).update(params) }
+      steps.all? do |step_class|
+        step = step_class.new(self)
+        result = step.update(params)
+
+        next true if step.complete?
+
+        result
```

Comment:
does it ever make sense to return result? Wondering if this should just be

```
steps.all? do |step_class|
  step = step_class.new(self)
  step.update(params)
  step.complete?
end
```

Reply:
yup, that's much more concise and does the same thing